util/test/sha3_256_tests.cpp

```diff
@@ -0,0 +1,73 @@
+// Concord
+//
+// Copyright (c) 2019 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the
+// LICENSE file.
+
+#include "gtest/gtest.h"
+#include "sha3_256.h"
+
+using namespace concord::util;
+
+namespace {
+
+// Convert a lowercase string hash to an array
+std::array<uint8_t, SHA3_256::SIZE_IN_BYTES> string_to_array(const char* s) {
+  std::array<uint8_t, SHA3_256::SIZE_IN_BYTES> hash = {0};
+
+  for (size_t i = 0; i < 64; i++) {
+    auto c = s[i];
+    Assert((c >= '0' && c <= '9') || (c >= 'a' && c <= 'f'));
+    uint8_t nibble = 0;
+    ;
+    if (c >= '0' && c <= '9') {
+      nibble = c - '0';
+    } else if (c >= 'a' && c <= 'f') {
+      nibble = c - 'a' + 10;
+    }
+
+    if (i % 2 == 0) {
+      // upper nibble
+      hash[i / 2] = nibble << 4;
+    } else {
+      // lower nibble
+      hash.at(i / 2) |= nibble;
+    }
+  }
+
+  return hash;
+}
+
+TEST(sha3_256_test, basic) {
+  auto sha3 = SHA3_256();
+  auto expected = string_to_array("a7ffc6f8bf1ed76651c14756a061d662f580ff4de43b49fa82d80a4b80f8434a");
```

Comment:
I think a small comment explaining how you computed the expected hashes in this test might be a good idea - to help others verify that these really are the SHA3-256 hashes of these inputs, and not just the output of this implementation.

Reply:
Good call!.  Fixed.